Templates/Common-Backend-Scripts/utilities/fetchBuildDependencies.groovy

```diff
@@ -0,0 +1,421 @@
+import groovy.util.*
+import java.nio.file.*
+
+
+// script properties
+@Field Properties props = new Properties()
+@Field def artifactRepositoryHelpers // Helpers to download
+@Field def applicationDescriptorUtils // Helper to parse Application Descriptor
+@Field def packageBuildOutputs // Helpers to download
+
+// Parse arguments from command-line
+parseArgs(args)
+
+// Print parms
+println("** Script configuration:")
+props.sort().each { k,v->
+	if ( k == "artifactRepository.password" )
+		println "   $k -> xxxxxx "
+	else
+		println "   $k -> $v"
+}
+
+// Load and verify helpers
+def scriptDir = new File(getClass().protectionDomain.codeSource.location.path).parent
+
+// Artifact RepositoryHelpers are used to download the package
+File artifactRepositoryHelpersScriptFile = new File("${scriptDir}/../../../Pipeline/PackageBuildOutputs/ArtifactRepositoryHelpers.groovy")
+if (artifactRepositoryHelpersScriptFile.exists()) {
+	artifactRepositoryHelpers = loadScript(artifactRepositoryHelpersScriptFile)
+} else {
+	println("*! [ERROR] The Artifact Repository Helper script '${props.artifactRepositoryHelpersScript}' doesn't exist. Exiting.")
+	System.exit(8)
+}
+
+File applicationDescriptorUtilsFile = new File("${scriptDir}/../../../Pipeline/PackageBuildOutputs/utilities/applicationDescriptorUtils.groovy")
+if (applicationDescriptorUtilsFile.exists()) {
+	applicationDescriptorUtils = loadScript(applicationDescriptorUtilsFile)
+} else {
+	println("*! [ERROR] The Application Descriptor Helper script '$applicationDescriptorUtilsFile.getName()' was not found. Exiting.")
+	System.exit(8)
+}
+
+File applicationDescriptorFile = new File(props.applicationDescriptor)
+if (!applicationDescriptorFile.exists()) {
+	println("*! [ERROR] The Application Descriptor file '${applicationDescriptorFile.getName()}' doesn't exist. Exiting.")
+	System.exit(8)
+}
+
+// setup import directory
+File importFolder = new File("$props.workspace/imports")
+File baselineFolder = new File("$props.workspace/baseline")
+if (importFolder.exists()) importFolder.deleteDir()
+importFolder.mkdirs()
+
+// setup package cache
+tmpPackageDir = (props.enablePackageCache && props.enablePackageCache.toBoolean() && props.packageCacheLocation) ? new File(props.packageCacheLocation) : new File("$props.workspace/imports_download/")
+if (!tmpPackageDir.exists()) tmpPackageDir.mkdirs()
+
+// Parse the application descriptor and application configurations based on the defined schema
+applicationDescriptor = applicationDescriptorUtils.readApplicationDescriptor(applicationDescriptorFile)
+
+
+ArrayList<ExternalDependency> externalDependencies = new ArrayList<>()
+
+if (applicationDescriptor.dependencies) {
+
+	println("*** Fetching external application packages")
+
+	// Loop through all dependencies found in AD
+	applicationDescriptor.dependencies.each { dependency ->
+
+		// validate dependency record
+		assert dependency.type : "Missing dependency type attribute in dependency record"
```

Comment:
Can we avoid using asserts? I feel it doesn't help control correctly the execution. Would rather go with if/then/else blocks and proper error management.